tests/unit_tests/test_matchmaker_queue.py

```diff
@@ -420,11 +418,32 @@ async def find_matches():
     )
 
 
+@pytest.mark.asyncio
+async def test_queue_cancel_while_being_matched_registers_failed_attempt(
+        matchmaker_queue, matchmaker_players_all_match
+    ):
+    p1, p2, p3, p4,  _ = matchmaker_players_all_match
+    searches = [Search([p1]), Search([p2]), Search([p3]), Search([p4])]
+    for search in searches:
+        asyncio.create_task(matchmaker_queue.search(search))
+
+    searches[0].cancel()
+
+    await asyncio.sleep(0.01)
+    await matchmaker_queue.find_matches()
+
+    for search in searches[1:]:
+        assert search.is_matched ^ (search.failed_matching_attempts == 1)
```

Comment:
WTF!! Using xor for something that's not encryption???? My brain hurts, I'm not used to thinking about xor as an actual logical operator xD

Reply:
xD At least in python you can do xor of booleans here unlike in certain other languages where you have to write `search.is_matched != (search.failed_matching_attempts == 1)` for an xor of two booleans.
I understand that they're the same thing, but they somehow read very differently...